lua/lazyvim/plugins/extras/lang/python.lua

```diff
@@ -0,0 +1,116 @@
+return {
+  {
+    "nvim-treesitter/nvim-treesitter",
+    opts = function(_, opts)
+      if type(opts.ensure_installed) == "table" then
+        vim.list_extend(opts.ensure_installed, { "ninja", "python", "rst", "toml" })
+      end
+    end,
+  },
+  {
+    "jose-elias-alvarez/null-ls.nvim",
+    opts = function(_, opts)
+      local nls = require("null-ls")
+      opts.sources = opts.sources or {}
+      vim.list_extend(opts.sources, {
+        -- Order of formatters matters. They are used in order of appearance.
+        nls.builtins.formatting.ruff.with({ filetypes = { "python" }, prefer_local = ".venv/bin" }),
+        nls.builtins.formatting.black.with({ filetypes = { "python" }, prefer_local = ".venv/bin" }),
+      })
+    end,
+    dependencies = {
+      "mason.nvim",
+      opts = function(_, opts)
+        opts.ensure_installed = opts.ensure_installed or {}
+        vim.list_extend(opts.ensure_installed, { "black", "ruff" })
+      end,
+    },
+  },
+  {
+    "neovim/nvim-lspconfig",
+    opts = {
+      servers = {
+        pyright = {
+          settings = {
+            pyright = {
+              disableLanguageServices = false,
+              disableOrganizeImports = false,
+            },
+            python = {
+              analysis = {
+                autoImportCompletions = true,
+                autoSearchPaths = true,
+                diagnosticMode = "workspace",
+                typeCheckingMode = "basic",
+                useLibraryCodeForTypes = true,
+              },
+            },
+          },
```

Comment:
Just wondering, what's the benefit of explicitly setting the defaults for these?

Reply:
Not every setting is set by default. The neovim-lspconfig only set these settings:

```lua
{
  python = {
    analysis = {
      autoSearchPaths = true,
      diagnosticMode = "workspace",
      useLibraryCodeForTypes = true
    }
  }
}
```

I often set some values explicitly because of better control. If you thinking it would be better to leave the decision to lower level package I will change this. I am new to neovim ecosystem and I would really appreciate the feedback.

Reply:
> Not every setting is set by default. The neovim-lspconfig only set these settings:

I think the rest are the Pyright server defaults, just not set by nvim-lspconfig.

> I often set some values explicitly because of better control. If you thinking it would be better to leave the decision to lower level package I will change this.

I think this is fine either way. I'm just wary of explicitly setting defaults in general because I think it's a double-edged sword. Although it increases visibility and control, it can also obscure upstream default changes.

Reply:
yes, please don't explicitely set defaults